docs/design/git-submodules.md

```diff
@@ -137,7 +137,74 @@ I.e. outcomes we would like to see if there were no constraints whatsoever.
 
 ### Guiding principles
 
-TODO
+These guiding principles exist to make it easy for us to create a coherent user
+experience, and for users to understand how submodules work, especially when
+they diverge from Git's implementation.
+
+#### Submodules are not standalone repositories
+
+Treating the submodule as a standalone jj repository is both detrimental and
+unnecessary. The reasons why are easiest understood in comparison to Git, where
+submodules _are_ standalone Git repositories:
+
+- As a standalone repository, it is possible to bypass the superproject to
+  interact with a submodule. This means that properties that the superproject
+  relies on may be invalidated unexpectedly.
+
+  For example, when `git gc` attempts to delete 'unreachable' objects, it takes
+  into account the objects reachable by refs in the repository. However, in a
+  submodule, it is possible that a commit is not reachable by a submodule ref,
+  but it is reachable by a superproject commit. Without knowledge of the
+  superproject, the submodule may delete objects that the superproject is
+  relying on ([see relevant StackOverflow
+  question](https://stackoverflow.com/questions/31640270/will-git-garbage-collect-commit-in-submodule-referred-to-by-a-top-level-reposito)).
+
+- With Git submodules, 'the repository' changes based on where `git` is run in
+  the working tree, which is a source of
+  [confusion](https://github.com/martinvonz/jj/issues/494#issuecomment-1404338917).
+
+We should keep in mind that a submodule exists to be integrated with a
+superproject, otherwise, the submodule could just be an independent clone. As
+such, jj should enforce tight integrations between superproject and submodule
+tightly by requiring that all interactions with the submodule must be initiated
+from the superproject.
+
+#### Commands should involve submodules by default
+
+Git does not recurse into submodules by default (it requires the
+`--recurse-submodules` flag or the `submodule.recurse` config setting), and this
+is a constant source of confusion for users, e.g.:
+
+- Non-recursive `git clone` will not clone submodules, resulting in their
+  contents being unpopulated in the working tree.
+
+- Non-recursive `git checkout` does not update the submodule index or working
+  tree, resulting in the submodule being 'dirty' and the working tree being out
+  of sync with the superproject.
+
+We should always consider submodules. If they are ignored, we should be explicit
+to the user.
+
+#### Submodules are globally managed
+
+In a regular jj or Git repository, objects are reusable because they are stored
+in a central database, regardless of the working copy. Similarly, we should
```

Comment:
nit: "central" here confused me at first because it thought of it in the "centralized VCS" sense. Maybe "in a database that's independent of the working copy" or something like that?